pyccl/halos/pk_4pt.py

```diff
@@ -495,6 +500,81 @@ def _allocate_profiles(prof, prof2, prof3, prof4, prof12_2pt, prof34_2pt):
     return prof, prof2, prof3, prof4, prof12_2pt, prof34_2pt
 
 
+def _allocate_profiles1pt(prof, prof2, prof3, prof4):
+    if not isinstance(prof, HaloProfile):
```

Comment:
I think we don't need to check the type (we stopped doing this), since it'll fail anyway with an equally useful python-level message

Reply:
How is this function different to the already-existing `_allocate_profiles` above it?

Reply:
For the 4h case, we only need the profiles, without the 2pt ones.
There is another difference. in the already existing `_allocate_profiles` we have
```
if prof4 is None:
    prof4 = prof2
```
whereas in the `_allocate_profiles1pt` has `prof4 = prof3`, which I think makes more sense. However, changing this in `_allocate_profiles` breaks a tkssc test. Which one should I leave?

Reply:
We need to choose one default behaviour and stick to it in all these functions. I actually think the behaviour in `_allocate_profiles` makes a tiny bit more sense. Since this is also what's used by the 1h and SSC terms, which are already part of the API, I suggest we keep that (otherwise we break the API anyway).

Reply:
and going back to my original point, to avoid rewriting code, in that case, simply call `_allocate_profiles1pt` inside `_allocate_profiles`

Reply:
Done